### PR TITLE
Create edit servlet.

### PR DESCRIPTION
**Pull request by basaam0**
Servlet called by receipt analysis page to edit the receipt categories, store name, price, and/or transaction date.

### CODE REVIEW

File: src/main/java/servlets/EditReceiptServlet.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    if (!userService.isUserLoggedIn()) {
+      logger.warning(USER_NOT_LOGGED_IN_WARNING);
+      response.setStatus(HttpServletResponse.SC_FORBIDDEN);
+      response.getWriter().println(USER_NOT_LOGGED_IN_WARNING);
+      return;
+    }
+
+    Entity receipt;
+    try {
+      long id = Long.parseLong(request.getParameter("id"));
```
Comment (avagarcia0) [inline src/main/java/servlets/EditReceiptServlet.java:78]: If you want the receipt analysis page to send an ID, you'll have to modify upload.js to send the ID when it redirects to the receipt analysis page.

Comment (basaam0) [reply src/main/java/servlets/EditReceiptServlet.java:78]: Added ID in #88.

File: src/main/java/servlets/EditReceiptServlet.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      receipt = createUpdatedReceipt(request, id);
+    } catch (EntityNotFoundException | InvalidPriceException | InvalidDateException
+        | NumberFormatException e) {
```
Comment (emilyvera) [inline src/main/java/servlets/EditReceiptServlet.java:81]: Maybe use a better name that just "e"

File: src/main/java/servlets/EditReceiptServlet.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      logger.warning(e.toString());
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      response.getWriter().println(e.toString());
+      return;
+    }
+
+    datastore.put(receipt);
```
Comment (emilyvera) [inline src/main/java/servlets/EditReceiptServlet.java:88]: Should catch DatastoreFailureException here

**Pull request merged**